File: Concreta.java
public class AudioPlayer
{
    protected Boolean isOpen = false;
    protected Boolean isPlaying = false;
    private float volume;

    public AudioPlayer()
    {
        System.out.println("The AudioPlayer constructor was invoked.");
    }

    public void open(String filePath)
    {
        this.isOpen = true;
        System.out.println("The audiofile: " + filePath + " is open.");
    }

    public void play()
    {
        if(isOpen) 
            this.isPlaying = true;
            System.out.println("The audiofile is playing.");
    }

    public void stop()  
    {

    if(isPlaying) 
        this.isPlaying = false;
        System.out.println("The audiofile is stopped.");
    }
    public void setVolume(float value)
    {
        this.volume = value;
        System.out.println("The volume value is: " + volume);
    }

};


class VLC  extends AudioPlayer
{

    private float pitch;
    public VLC()
    {
    System.out.println("The VLC constructor was invoked.");
    }

    public void setPitch(float value)
    {
        this.pitch = value;
        System.out.println("The pitch value is: " + pitch);
    }

}

public static void main(String[] args)
{
    AudioPlayer player = new AudioPlayer();
    player.open("./resources/orchestral.ogg");
    player.play();
    player.setVolume(4);
    System.out.println("------");
    
    VLC vlcPlayer = new VLC();
    vlcPlayer.open("./resources/orchestral.ogg");
    vlcPlayer.play();
    vlcPlayer.setVolume(13);
    System.out.println("---");

} 

File: Concreta.py

class AudioPlayer:
    def __init__(self):
        print("The AudioPlayer constructor was invoked.")
        self.isOpen = False
        self.isPlaying = False
        self.volume = 0
    
    def open(self, filePath):
        self.filePath = filePath
        self.isOpen = True
        print("The audiofile: " + self.filePath + " is open.")

    def play(self):
        if self.isOpen:
            self.isPlaying = True
            print("The audiofile is playing.")
        
    def stop(self):
        if self.isPlaying:
            self.isPlaying = False
            print("The audiofile is stopped.")
    
    def setVolume(self, value):
        self.volume = value
        print("The volume value is: " + self.volume)

#****************************************************************************
# ------------Clase Hija o derivada----------------
 
class VLC(AudioPlayer):
    print("The VLC constructor was invoked.")

    def setPitch(self, value):
        self.pitch = value
        print("The pitch value is: " + self.pitch)

#****************************************************************************
# ----instanciacion de clase y sus funciones miembro--
player = AudioPlayer()
player.open("./resources/orchestral.ogg")
player.play()
player.setVolume(13)
print("--"*20)
vlcPlayer = VLC()
vlcPlayer.open("./resources/orchestral.ogg")
vlcPlayer.play()
vlcPlayer.setVolume(13)